public/app/features/explore/ShortLinkButtonMenu.tsx

```diff
@@ -0,0 +1,99 @@
+import { css } from '@emotion/css';
+import React, { useState } from 'react';
+
+import { GrafanaTheme2, urlUtil } from '@grafana/data';
+import { Stack } from '@grafana/experimental';
+import { reportInteraction } from '@grafana/runtime';
+import { useStyles2, ToolbarButton, Dropdown, Menu } from '@grafana/ui';
+import { t } from 'app/core/internationalization';
+import { createAndCopyShortLink } from 'app/core/utils/shortLinks';
+import { useSelector } from 'app/types';
+
+import { getUrlStateFromPaneState } from './hooks/useStateSync';
+import { selectPanes } from './state/selectors';
+
+export function ShortLinkButtonMenu() {
+  const styles = useStyles2(getStyles);
+  const panes = useSelector(selectPanes);
+  const [isOpen, setIsOpen] = useState(false);
+  const onCopyShortLink = (url?: string) => {
+    createAndCopyShortLink(url || global.location.href);
+    reportInteraction('grafana_explore_shortened_link_clicked');
+  };
+
+  const MenuActions = () => {
+    return (
+      <Menu>
+        <Menu.Item
+          key="copy-link"
+          label="Copy shortened URL"
+          icon="link"
+          onClick={() => onCopyShortLink()}
+          className={styles.menuItem}
+        />
+        <Menu.Item
+          key="copy-link-abs-time"
+          label="Copy absolute time URL"
+          icon="clock-nine"
+          description="Locks the current time for consistent viewing"
+          onClick={() => {
+            const exploreIds = Object.keys(panes);
+            const urlStates = Object.values(panes)
+              .map((pane, i) => {
+                if (pane) {
+                  const urlState = getUrlStateFromPaneState(pane);
+                  urlState.range = {
+                    to: pane.range.to.valueOf().toString(),
+                    from: pane.range.from.valueOf().toString(),
+                  };
+                  return JSON.stringify({ [exploreIds[i]]: { ...urlState } });
+                } else {
+                  return undefined;
+                }
+              })
+              .filter((urlState): urlState is string => urlState !== undefined);
+            onCopyShortLink(urlUtil.renderUrl('/explore', { panes: urlStates, schemaVersion: 1 }));
+          }}
+          className={styles.menuItem}
+        />
+      </Menu>
+    );
+  };
+
+  // we need the Toolbar button click to be an action separate from opening/closing the menu
+  return (
+    <Stack gap={0} direction="row" alignItems="baseline" wrap={false}>
+      <ToolbarButton
```

Comment:
maybe we should wrap this in a `ToolbarButtonRow` so we get nice behaviours automatically when users resize their window:

without:
With `ToolbarButtonRow`:
![Screenshot 2023-11-02 at 11 15 11](https://github.com/grafana/grafana/assets/1170767/628b49bf-e9d4-4571-a32a-ec0b3ce40d87)

without `ToolbarButtonRow`:
![Screenshot 2023-11-02 at 11 15 02](https://github.com/grafana/grafana/assets/1170767/99212d9b-4e8c-4f7e-94d4-60027d52920e)

Reply:
Fixed! (this is not in the ephemeral, I missed this until later)